static/js/compose.js

```diff
@@ -475,13 +475,25 @@ function validate_stream_message_announce(stream_name) {
     return true;
 }
 
-function validate_stream_message_announcement_only(stream_name) {
-    // Only allow realm admins to post to announcement_only streams.
-    const is_announcement_only = stream_data.get_announcement_only(stream_name);
-    if (is_announcement_only && !page_params.is_admin) {
+function validate_stream_message_post_policy(stream_name) {
+    const stream_post_policy = stream_data.get_stream_post_policy(stream_name);
+    if (stream_post_policy === 2 && !page_params.is_admin) {
```

Comment:
These numerical checks would be much more legibly written as `stream_post_policy === stream_edit.stream_post_policy_values.admins.code`.  We generally try to avoid having any "magic numbers" anywhere in the codebase (since basically any one needs a comment explaining what it means).

Reply:
Should this change be made to the handlebars files too?
Made the changes in all other places.

Reply:
Should this change be made in the handlebars files too?
Made the change in all other files

Reply:
Ideally, yes: one should in theory be able to pass the data object into the templates.  It might be annoying to do, though, so if it's tricky, it's OK to have it there.

Reply:
Understood, I updated the templates